### PR TITLE
create a grpc client qps test logging reporter

### PR DESCRIPTION
**Pull request by apolcyn**
add an optional QPS test reporter that can send a benchmark `ScenarioResult` in an RPC.

cc @adelez @ctiller

### CODE REVIEW

**Comment by apolcyn** [comment]
just updated to fix the clang-format problem.

cc @yang-g after offline chat I think this would be useful to get in

File: test/cpp/qps/benchmark_config.cc [hunk 1]
```diff
@@ -75,6 +83,13 @@ static std::shared_ptr<Reporter> InitBenchmarkReporters() {
     composite_reporter->add(std::unique_ptr<Reporter>(
         new JsonReporter("JsonReporter", FLAGS_scenario_result_file)));
   }
+  if (FLAGS_enable_rpc_reporter) {
+    GPR_ASSERT(strlen(FLAGS_rpc_reporter_server_address.c_str()) > 0);
```
Comment (yang-g) [inline test/cpp/qps/benchmark_config.cc]: !FLAGS_rpc_reporter_server_address.empty() ?

File: src/proto/grpc/testing/services.proto [hunk 2]
```diff
@@ -33,6 +33,7 @@ syntax = "proto3";
 
 import "src/proto/grpc/testing/messages.proto";
 import "src/proto/grpc/testing/control.proto";
+import "src/proto/grpc/testing/stats.proto";
```
Comment (yang-g) [inline src/proto/grpc/testing/services.proto]: Do you need to update the BUILD file for this?

Comment (apolcyn) [inline src/proto/grpc/testing/services.proto]: woops done

File: test/cpp/qps/benchmark_config.cc (hunk 1 above)
Comment (apolcyn) [inline test/cpp/qps/benchmark_config.cc]: done

**Comment by apolcyn** [comment]
* asan c++ pr: failed on known issue(https://grpc-testing.appspot.com/job/gRPC_pull_requests_asan_cpp/3690/), in poll, epoll, poll-cv poll stratgey test configs
* macos pr: https://github.com/grpc/grpc/issues/10539
* windows PR: https://github.com/grpc/grpc/issues/9088
* protability pr: https://grpc-testing.appspot.com/job/gRPC_portability_pull_requests_linux/4441/
* interop pr: https://github.com/grpc/grpc/issues/10525

**Comment by apolcyn** [comment]
latest update since review added a missing comma in modification to `src/proto/grpc/testing/BUILD`, (previous tests failed on bazel basic tests)

**Pull request merged**